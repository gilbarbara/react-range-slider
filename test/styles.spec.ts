import getStyles from '../src/styles';

describe('styles', () => {
  describe('without custom styles', () => {
    it('should return a RangeSliderStyles', () => {
      expect(getStyles()).toMatchSnapshot();
    });
  });

  describe('with custom styles', () => {
    it('should return a modified RangeSliderStyles', () => {
      expect(
        getStyles({
          options: {
            thumbBorderRadius: 0,
            trackBorderRadius: 0,
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
