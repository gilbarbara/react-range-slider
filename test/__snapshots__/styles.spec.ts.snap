// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles with custom styles should return a modified RangeSliderStyles 1`] = `
Object {
  "options": Object {
    "thumbBorderRadius": 0,
    "trackBorderRadius": 0,
  },
  "rail": Object {
    "boxSizing": "border-box",
    "height": "20px",
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": "20px",
  },
  "rangeX": Object {
    "backgroundColor": "#007bff",
    "borderRadius": 0,
    "height": "100%",
    "position": "absolute",
    "top": 0,
  },
  "rangeXY": Object {
    "backgroundColor": "#007bff",
    "borderRadius": 0,
    "bottom": 0,
    "position": "absolute",
  },
  "rangeY": Object {
    "backgroundColor": "#007bff",
    "borderRadius": 0,
    "bottom": 0,
    "left": 0,
    "position": "absolute",
    "width": "100%",
  },
  "sliderX": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": 32,
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": "100%",
  },
  "sliderXY": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": "100%",
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": "100%",
  },
  "sliderY": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": "100%",
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": 32,
  },
  "thumbX": Object {
    "backgroundColor": "#fff",
    "border": "2px solid #000",
    "borderRadius": 0,
    "boxSizing": "border-box",
    "display": "block",
    "height": 26,
    "left": -5,
    "position": "absolute",
    "top": -3,
    "transition": "height 0.4s, width 0.4s",
    "width": "10px",
  },
  "thumbXY": Object {
    "backgroundColor": "transparent",
    "border": "2px solid #000",
    "borderRadius": "50%",
    "bottom": -10,
    "boxSizing": "border-box",
    "display": "block",
    "height": "20px",
    "left": -10,
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": "20px",
  },
  "thumbY": Object {
    "backgroundColor": "#fff",
    "border": "2px solid #000",
    "borderRadius": 0,
    "bottom": -5,
    "boxSizing": "border-box",
    "display": "block",
    "height": "10px",
    "left": -3,
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": 26,
  },
  "trackX": Object {
    "backgroundColor": "#ccc",
    "borderRadius": 0,
    "boxSizing": "border-box",
    "height": "20px",
    "position": "relative",
    "width": "100%",
  },
  "trackXY": Object {
    "backgroundColor": "#ccc",
    "borderRadius": 0,
    "boxSizing": "border-box",
    "height": "100%",
    "minHeight": "50px",
    "position": "relative",
    "width": "100%",
  },
  "trackY": Object {
    "backgroundColor": "#ccc",
    "borderRadius": 0,
    "boxSizing": "border-box",
    "height": "100%",
    "minHeight": "50px",
    "position": "relative",
    "width": "20px",
  },
}
`;

exports[`styles without custom styles should return a RangeSliderStyles 1`] = `
Object {
  "rail": Object {
    "boxSizing": "border-box",
    "height": "20px",
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": "20px",
  },
  "rangeX": Object {
    "backgroundColor": "#007bff",
    "borderRadius": "3px",
    "height": "100%",
    "position": "absolute",
    "top": 0,
  },
  "rangeXY": Object {
    "backgroundColor": "#007bff",
    "borderRadius": "3px",
    "bottom": 0,
    "position": "absolute",
  },
  "rangeY": Object {
    "backgroundColor": "#007bff",
    "borderRadius": "3px",
    "bottom": 0,
    "left": 0,
    "position": "absolute",
    "width": "100%",
  },
  "sliderX": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": 32,
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": "100%",
  },
  "sliderXY": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": "100%",
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": "100%",
  },
  "sliderY": Object {
    "boxSizing": "border-box",
    "display": "inline-block",
    "height": "100%",
    "padding": "6px",
    "transition": "height 0.4s, width 0.4s",
    "width": 32,
  },
  "thumbX": Object {
    "backgroundColor": "#fff",
    "border": "2px solid #000",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "display": "block",
    "height": 26,
    "left": -5,
    "position": "absolute",
    "top": -3,
    "transition": "height 0.4s, width 0.4s",
    "width": "10px",
  },
  "thumbXY": Object {
    "backgroundColor": "transparent",
    "border": "2px solid #000",
    "borderRadius": "50%",
    "bottom": -10,
    "boxSizing": "border-box",
    "display": "block",
    "height": "20px",
    "left": -10,
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": "20px",
  },
  "thumbY": Object {
    "backgroundColor": "#fff",
    "border": "2px solid #000",
    "borderRadius": "4px",
    "bottom": -5,
    "boxSizing": "border-box",
    "display": "block",
    "height": "10px",
    "left": -3,
    "position": "absolute",
    "transition": "height 0.4s, width 0.4s",
    "width": 26,
  },
  "trackX": Object {
    "backgroundColor": "#ccc",
    "borderRadius": "3px",
    "boxSizing": "border-box",
    "height": "20px",
    "position": "relative",
    "width": "100%",
  },
  "trackXY": Object {
    "backgroundColor": "#ccc",
    "borderRadius": "3px",
    "boxSizing": "border-box",
    "height": "100%",
    "minHeight": "50px",
    "position": "relative",
    "width": "100%",
  },
  "trackY": Object {
    "backgroundColor": "#ccc",
    "borderRadius": "3px",
    "boxSizing": "border-box",
    "height": "100%",
    "minHeight": "50px",
    "position": "relative",
    "width": "20px",
  },
}
`;
